ecl/hthor/hthor.cpp

```diff
@@ -8414,6 +8437,32 @@ unsigned __int64 CHThorDiskReadBaseActivity::getLocalFilePosition(const void * r
 
 void CHThorDiskReadBaseActivity::closepart()
 {
+    if (opened && inputfileio && ldFile && partNum > 0)
+    {
+        unsigned previousPartNum = partNum-1;
+        if (previousPartNum < ldFile->numParts())
+        {
+            stat_type curDiskReads = inputfileio->getStatistic(StNumDiskReads);
+            IDistributedFile * dFile = ldFile->queryDistributedFile();
+            if (dFile)
+            {
+                if (superfile)
+                {
+                    unsigned subfile, lnum;
+                    if (superfile->mapSubPart(previousPartNum, subfile, lnum))
+                    {
+                        IDistributedSuperFile *super = dFile->querySuperFile();
+                        dFile = &(super->querySubFile(subfile, true));
+                    }
+                }
+                dFile->addAttrValue("@numDiskReads", curDiskReads);
```

Comment:
@jakesmith is there any issue with updating these properties?  Is the dali entry locked, and does it happen straight away.  What implications does this have once esp provides the meta information!

Reply:
addAttrValue automatically locks the file attr properties, for an atomic update, and the update is pushed immediately.
There will need to be an implemtation of this that does the same atomic update when it's via a esp service.

Reply:
Yes, I added a subtask/comment to the jira to make sure it was covered.